classes/Product.php

```diff
@@ -4443,6 +4443,28 @@ public static function duplicateTags($id_product_old, $id_product_new)
 
         return Db::getInstance()->insert('product_tag', $data);
     }
+    
+    public static function duplicateTaxes($id_product_old, $id_product_new)
+    {
+        $query = new DbQuery();
+        $query->select('id_tax_rules_group, id_shop');
+        $query->from('product_shop');
+        $query->where('`id_product` = ' . (int) $id_product_old);
+
+        $results = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($query->build());
+
+        if (!empty($results)) {
+            foreach ($results as $result) {
+                Db::getInstance()->update(
```

Comment:
We can maybe check if something went wrong and return false if a problem happened?